packages/account/src/utils/Types.ts

```diff
@@ -70,8 +83,10 @@ export interface BiconomySmartAccountV2Config extends BaseSmartAccountConfig {
   implementationAddress?: string;
   defaultFallbackHandler?: string;
   rpcUrl?: string; // as good as Provider
+  signer: Signer;
```

Comment:
Could use typescript union types here:

```
type ConditionalValidationProps = {
	defaultValidationModule?: BaseValidationModule;
	signer: Signer;
} | {
	defaultValidationModule: BaseValidationModule;
	signer?: Signer;
}

export interface BaseSmartAccountConfig extends ConditionalValidationProps {
	...
}
```

Reply:
did you apply this @GabiDev45 ?

Reply:
@livingrockrises this is what I was talking about yesterday when I said conditional types, I remember trying something similar once but for some reason it didnt work as expected, will check

Reply:
I think this won't work tho as interfaces cannot extend union types, we either make both signer and defaultValidationModule optional or we turn BaseSmartAccountConfig into a type instead of interface

Reply:
You can't make BaseSmartAccountConfig a type instead?

Reply:
Oh you did, nice